lib/keytar.js

```diff
@@ -0,0 +1,82 @@
+var keytar = require('../build/Release/keytar.node')
+
+function checkRequired(val, name) {
+  if (!val || val.length <= 0) {
+    throw new Error(name + ' is required.');
+  }
+}
+
+function callbackOrPromise(maybeCallback) {
+  if (maybeCallback && typeof maybeCallback === 'function') {
+    return {
+      value: null,
+      callback: function(err, val) {
+        if (err) {
+          maybeCallback(err)
+        } else {
+          val === undefined ? maybeCallback(err) : maybeCallback(err, val)
+        }
+      }
+    }
+  } else if (maybeCallback) {
+    throw new Error('Callback must be a function.')
+  } else {
+    var res, rej;
+    var promise = new Promise(function(resolve, reject) {
+      res = resolve
+      rej = reject
+    })
+    return {
+      value: promise,
+      callback: function (err, val) {
+        if (err) {
+          rej(err)
+        } else {
+          res(val)
+        }
+      }
+    }
+  }
+}
+
+module.exports = {
+  getPassword: function (service, account, callback) {
+    checkRequired(service, 'Service')
+    checkRequired(account, 'Account')
+
+    var ret = callbackOrPromise(callback)
```

Comment:
Yeah, I think I am ok with either too honestly. Maybe, given that we are going to use these new methods via the Promise based API, we could just expose that and save an object allocation?